lib/fabrik/DirectorManager.js

```diff
@@ -223,10 +240,58 @@ class DirectorManager extends BaseManager {
       });
   }
 
+  executeActions(phase, context) {
+    return Promise.try(() => {
+      let serviceLevelActions = this.service.actions;
+      let planLevelActions = phase === CONST.SERVICE_LIFE_CYCLE.PRE_UPDATE ? catalog.getPlan(context.params.previous_values.plan_id).actions :
+        this.plan.actions;
+      if (serviceLevelActions || planLevelActions) {
+        const cumilativeActions = serviceLevelActions ? (planLevelActions ? `${serviceLevelActions},${planLevelActions}` : serviceLevelActions) :
+          planLevelActions;
+        const actionsToPerform = _.chain(cumilativeActions)
+          .replace(/\s*/g, '')
+          .split(',')
+          .value();
+        if (actionsToPerform.length === 0) {
+          logger.info(`no actions to perform for ${context.deployment_name}`);
+          return {};
+        }
+        logger.info(`actionsToPerform - @service - ${serviceLevelActions} , @plan - ${planLevelActions}`);
+        logger.info(`Cumulative actions to perform on ${context.deployment_name} - ${actionsToPerform}`);
+        const actionResponse = {};
+        let actionStartTime = Date.now();
+        let actionEndTime;
+        return Promise.map(actionsToPerform, (action) => {
+            logger.debug(`Looking up action ${action}`);
+            const actionHandler = ActionManager.getAction(phase, action);
+            _.assign(context, {
+              'instance_guid': this.getInstanceGuid(context.deployment_name)
+            });
+            return actionHandler(context)
+              .tap(resp => actionResponse[action] = resp);
+          })
+          .tap(() => logger.info(`${phase} response ...`, actionResponse))
+          .tap(() => {
+            actionEndTime = Date.now();
+            if (actionEndTime - actionStartTime > config.deployment_action_timeout) {
```

Comment:
So if deploy hooks fail, we do not go ahead with the deployment and fails the flow. Right?

Reply:
yes. It will throw an exception and the same will get populated and thrown to cf resulting in breaking the flow.
In case of long running scripts, if scripts take more than configured `deployment_action_timeout` then also we throw an exception and deployment doesn't go through

Reply:
May be you can add a TODO comment here that we could as well kill the process running the script. currently we let the script run and only validate post script run. May be we can improve on this. So just add a TODO: in the code with this comment, so that if someone else works on this is aware that we have thought thru this.